Automattic-Tracks-iOSTests/Test Helpers/Mocks.swift

```diff
@@ -0,0 +1,107 @@
+import Foundation
+import Sodium
+@testable import AutomatticTracks
+
+typealias LogFileCallback = (LogFile) -> ()
+typealias ErrorWithLogFileCallback = (Error, LogFile) -> ()
+
+enum MockError: Error {
+    case generic
+}
+
+class MockEventLoggingDataSource: EventLoggingDataSource {
+
+    var loggingEncryptionKey: String = "foo"
+    var previousSessionLogPath: URL? = nil
+
+    /// Overrides for logUploadURL
+    var _logUploadURL: URL = URL(string: "example.com")!
+    var logUploadURL: URL {
+        return _logUploadURL
+    }
+
+    func setLogUploadUrl(_ url: URL) {
+        self._logUploadURL = url
+    }
+
+    func withEncryptionKeys() -> Self {
+        let keyPair = Sodium().box.keyPair()!
+        loggingEncryptionKey = Data(keyPair.publicKey).base64EncodedString()
+        return self
+    }
+}
+
+class MockEventLoggingDelegate: EventLoggingDelegate {
```

Comment:
Very similar to my last comment, here is how I'd have written this class: c3debf184004dddb662de750d4893d6c02f00c12